api/envoy/config/metrics/v2/stats.proto

```diff
@@ -56,6 +57,76 @@ message StatsConfig {
   //
   // If not provided, the value is assumed to be true.
   google.protobuf.BoolValue use_all_default_tags = 2;
+
+  // Inclusion/exclusion matcher for stat name creation. If not provided, all stats are instantiated
+  // as normal. Preventing the instantiation of certain families of stats can improve memory
+  // performance for Envoys running especially large configs.
+  StatsMatcher stats_matcher = 3;
+}
+
+// Configuration for disabling stat instantiation.
+message StatsMatcher {
```

Comment:
I thought you mentioned that StringMatcher had other uses. Can we refactor the protos for those to reference this message?

Reply:
There's a `yaml` example in `api/envoy/type/matcher/metadata.proto`, and a use of `StringMatcher` in `api/envoy/config/rbac/v2alpha/rbac.proto` which uses a `StringMatcher`. I don't really want to redirect the docs for those to reference this message, which is really just a wrapper of ListStringMatcher with the added bias of being inclusive/exclusive. WDYT?

Reply:
Well my thinking is that one of the compelling reasons for using StringMatcher is that its UI could be shared across a number of config settings.

Is there no central proto file where this kind of shared UI can live? Could we add one? @mattklein123 what's your take?

In any case this refactoring -- if we decide to do it -- could be a follow-up.

Reply:
Agreed, putting a TODO in for myself to refactor this into a StringMatcherList.